docs/update-extension.md

```diff
@@ -0,0 +1,45 @@
+# Update OSM Arc extension
+
+## Supported Upgrades
+The OSM Arc extension can be upgraded up to the next minor version. Downgrades and major version upgrades are not supported at this time.
+
+## CRD Upgrades
+The OSM Arc extension cannot be upgraded to a new version if that version contains CRD updates without deleting the existing CRDs first. 
+
+Please refer to the [OSM CRD Upgrades documentation](https://github.com/openservicemesh/osm/blob/main/docs/upgrade_guide.md#crd-upgrades) to prep your cluster for such an upgrade. After deleting your CRDs per the OSM documentation, please follow the upgrade [instructions](#instructions) in this guide instead of using Helm or the OSM CLI.
+
+## OSM ConfigMap upgrades
+When upgrading, any custom edits to the OSM ConfigMap may be reverted to the default. 
+
+To ensure that these values are not overwritten, create a copy of the following JSON object and save it as `osm_extension.json`.
+
+```
+{
+  "properties": {
```

Comment:
Since this is user-facing doc, we should use `az k8s-extension --version <version to upgrade to> --configuration-settings`. 
The configuration file passed into this flag should only contain the non-default values. 

For the example below, this should be sufficient. 
```json
{
    "osm.OpenServiceMesh.envoyLogLevel" : "info"
}
```

Reply:
I moved these docs to a a dev folder. I will follow up in a separate PR for user facing docs.

Reply:
Moved these back. This repo will be used for dev scripts so using az rest is okay here